### PR TITLE
adds postgres version to the schema cache.

### PR DESCRIPTION
**Pull request by harshit54**
Adds the postgres version in the schema cache using SQL from the `pg-meta` library.

### CODE REVIEW

File: crates/pg_schema_cache/src/versions.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  current_setting('server_version_num') :: int8 AS version_num,
+  (
+    select
+      count(*) :: int8 AS active_connections
+    FROM
+      pg_stat_activity
+  ) AS active_connections,
+  current_setting('max_connections') :: int8 AS max_connections;"#
+        )
+        .fetch_all(pool)
```
Comment (psteinroe) [inline crates/pg_schema_cache/src/versions.rs:30]: why not just one row?

File: crates/pg_schema_cache/src/schema_cache.rs [hunk 2]
```diff
@@ -6,13 +6,15 @@ use crate::functions::Function;
...
 use crate::types::PostgresType;
+use crate::versions::Version;
 
 #[derive(Debug, Clone, Default)]
 pub struct SchemaCache {
     pub schemas: Vec<Schema>,
     pub tables: Vec<Table>,
     pub functions: Vec<Function>,
     pub types: Vec<PostgresType>,
+    pub versions: Vec<Version>,
```
Comment (psteinroe) [inline crates/pg_schema_cache/src/schema_cache.rs:17]: any reason to have a Vec instead of a single struct?

**Review by psteinroe — CHANGES_REQUESTED** [review]
Thanks for the contribution! Just one minor note 🙏🏼

**Comment by harshit54** [comment]
Hey @psteinroe , thanks for the reply.

While it does make sense to have a single type instead of a Vec, the `SchemaCacheItem` `trait` is defined as 

```rust
pub trait SchemaCacheItem {
    type Item;

    async fn load(pool: &PgPool) -> Vec<Self::Item>;
}
```
is defined as having a `Vec` of the type. Should we make the `version` implement a different trait?

**Comment by psteinroe** [comment]
Ah yeah I think it makes sense to refactor this a bit. Do you want to do it in your pr or should we merge and do a follow-up? 🙏🏼

**Comment by harshit54** [comment]
Let's do a follow up. I'm not sure how the refactoring will look like at the moment.

**Comment by psteinroe** [comment]
sounds good! the ci is failing due to an unrelated issue. I am going to fix it soon and then merge 🙏🏼

**Pull request merged**